server/services/src/test/java/org/zanata/service/impl/TranslationFileServiceImplTest.java

```diff
@@ -20,31 +20,153 @@
  */
 package org.zanata.service.impl;
 
+import com.google.common.base.Optional;
 import org.hibernate.Session;
 import org.jglue.cdiunit.InRequestScope;
 import org.junit.Test;
 import org.junit.runner.RunWith;
 import org.mockito.Mock;
 import org.zanata.ZanataTest;
 import org.zanata.common.DocumentType;
+import org.zanata.common.LocaleId;
+import org.zanata.common.ProjectType;
+import org.zanata.dao.DocumentDAO;
+import org.zanata.dao.ProjectIterationDAO;
+import org.zanata.exception.ZanataServiceException;
+import org.zanata.model.HDocument;
+import org.zanata.model.HLocale;
+import org.zanata.model.HProject;
+import org.zanata.model.HProjectIteration;
+import org.zanata.rest.dto.resource.TranslationsResource;
 import org.zanata.test.CdiUnitRunner;
 
 import javax.enterprise.inject.Produces;
 import javax.inject.Inject;
 
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.InputStream;
+import java.nio.file.Files;
+
 import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.Assertions.fail;
+import static org.mockito.Mockito.when;
 
 @RunWith(CdiUnitRunner.class)
 public class TranslationFileServiceImplTest extends ZanataTest {
 
     @Inject
     TranslationFileServiceImpl transFileService;
 
+    @Mock
+    @Produces
+    private ProjectIterationDAO projectIterationDAO;
+
+    @Mock
+    @Produces
+    private DocumentDAO documentDAO;
+
     @Produces @Mock Session session;
 
     @Test
     @InRequestScope
     public void hasPlainTextAdapter() {
         assertThat(transFileService.hasAdapterFor(DocumentType.PLAIN_TEXT)).isTrue();
     }
+
+    @Test
+    @InRequestScope
+    public void parsePoInFileProject() throws Exception {
+        String project = "test";
+        String version = "master";
+        String docId = "test.pot";
+        String poContent = "msgid \"\"\n" +
+                "msgstr \"\"\n" +
+                "\"Project-Id-Version: test-master\\n\"\n" +
+                "\n" +
+                "#, c-format\n" +
+                "msgid \"Thing 1\"\n" +
+                "msgid_plural \"%d files removed\"\n" +
+                "msgstr[0] \"1 aoeuaouaou\"\n" +
+                "msgstr[1] \"%d aoeuaouao\"\n" +
+                "msgstr[2] \"\"";
```

Comment:
This test would be easier to read in Kotlin as a multi-line string.

Reply:
/runs